webextension/privileged/ui/CloudDownloadsView.jsm

```diff
@@ -0,0 +1,760 @@
+   * Registers Move Download option by a) creating a template moveDownload
+   * menuItem and moveDownloadSubMenu (if multiple providers) inside DownloadsContextMenu
+   * b) an event listener to listen to contextmenu event on downloadsListBox
+   * MenuItems are updated with label attributes before displaying Move Download option inside
+   * _displayMoveToCloudContextMenuItem
+   */
+  async registerContextMenu(browserWindow) {
+    try {
+      if (!browserWindow || !browserWindow.document) {
+        return;
+      }
+
+      // There is no other API that awaits the result of CloudStorageInternal.init()
+      // Invoke getDownloadFolder on CloudStorage API to ensure API is initialized
+      // before asking CloudStorage API for storage providers info
+      await CloudStorage.getDownloadFolder();
+      const providers = await CloudStorage.getStorageProviders();
+      this.providers = await this._filterProvidersMap(providers);
+
+      // Continue only if cloud providers exists on user device
+      if (this.providers.size === 0) {
+        return;
+      }
+
+      const aPopupMenu = browserWindow.document.getElementById("downloadsContextMenu");
+      if (!aPopupMenu) {
+        return;
+      }
+
+      // Get menuItem 'Copy Location', to insert skeleton 'Move download' menuitem and a menu separator after it
+      const menuItem = aPopupMenu.getElementsByAttribute("command", "downloadsCmd_copyLocation")[0];
+      if (!menuItem) {
+        return;
+      }
+
+      const fragment = browserWindow.document.createDocumentFragment();
+      const separator = browserWindow.document.createElement("menuseparator");
+      separator.setAttribute("id", "moveDownloadSeparator");
+      fragment.appendChild(separator);
+
+      if (this.providers.size > 1) {
+        const moveDownloadMenu = browserWindow.document.createElement("menu");
+        moveDownloadMenu.setAttribute("id", "moveDownload");
+        moveDownloadMenu.setAttribute("hidden", "true");
+        const moveDownloadMenuPopup = browserWindow.document.createElement("menupopup");
+        moveDownloadMenuPopup.setAttribute("id", "moveDownloadSubMenu");
+        moveDownloadMenuPopup.setAttribute("hidden", "true");
+
+        this.providers.forEach((value, key) => {
+          const moveDownloadItem = browserWindow.document.createElement("menuitem");
+          moveDownloadMenuPopup.appendChild(moveDownloadItem);
+        });
+
+        moveDownloadMenu.appendChild(moveDownloadMenuPopup);
+        fragment.appendChild(moveDownloadMenu);
+      } else {
+        // Add skeleton Move to menu item in context menu
+        const moveDownloadItem = browserWindow.document.createElement("menuitem");
+        moveDownloadItem.setAttribute("id", "moveDownload");
+        moveDownloadItem.setAttribute("hidden", "true");
+        fragment.appendChild(moveDownloadItem);
+      }
+      aPopupMenu.insertBefore(fragment, menuItem.nextSibling);
+      aPopupMenu.addEventListener("command", this);
```

Comment:
You need to still update the `removeEventListener` call to use `command`, too.

Reply:
done